roles/rsyslog/templates/global.j2

```diff
@@ -30,4 +30,7 @@ global(
 {% if rsyslog_max_message_size is defined %}
   maxMessageSize="{{ logging_max_message_size }}"
 {% endif %}
+{% if logging_preserve_fqdn | d() %}
```

Comment:
Since `logging_preserve_fqdn` is in the public API of the role (which are the parameters that users can set), it should be defined in `defaults/main.yml` as `logging_preserve_fqdn: false` - then it will always be defined and you can omit the `| d()`

Reply:
Ok, I have set it in defaults.

Reply:
But then again, should it be in globa defaults or should i change it to rsyslog_preserve_fqdn and put it in the defaults of the rsyslog role since it is rsyslog specific.

There are also other options in specified at the logger level that probably should be rsyslog specific.

Reply:
You are correct that this is an rsyslog specific config.  However, we broke that API convention about 4 years ago when we decided not to add fluentd support to the logging role, and ever since then, we've been adding rsyslog specific parameters to the top level defaults/main.yml - https://github.com/linux-system-roles/logging/blob/main/defaults/main.yml#L41 - starting here, most of these are rsyslog specific.

So I would prefer putting this in the global defaults as a `logging_` parameter, and if/when we ever refactor this role to support some logging provider other than rsyslog, we'll have to figure out how to do that.